projects/e2fsprogs/fuzz/ext2fs_image_read_write_fuzzer.cc

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <stddef.h>
+#include <stdint.h>
+#include <unistd.h>
+#include <fuzzer/FuzzedDataProvider.h>
+
+#include "ext2fs/ext2fs.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  static const char* pattern = "/dev/shm/ext2XXXXXX";
+  int fd;
+  char* fname;
+  FuzzedDataProvider stream(data, size);
+  const unsigned int fuzz_target = stream.ConsumeIntegral<unsigned int>() % 6;
```

Comment:
let's create a enum and use `ConsumeEnum` method of FDP for this (as well as in the switch below), that way it'll be less fragile if we ever need to change/extend it